public/index.html

```diff
@@ -1,46 +1,305 @@
+        (this.ball.x <= this.playerLeft.x + this.playerWidth
+          && this.ball.x >= this.playerLeft.x
+          && this.ball.y + this.ballSide >= this.playerLeft.y
+          && this.ball.y <= this.playerLeft.y + this.playerHeight)) {
+        this.gameSpeed *= 1.05 // increase the speed by 5% each time
+        if (this.ball.dx < 0) this.ball.dx = this.gameSpeed
+        else this.ball.dx = -this.gameSpeed
+      }
+
+      if (this.ball.y + this.ballSide >= this.height || this.ball.y <= 0) {
+        if (this.ball.dy < 0) this.ball.dy = this.gameSpeed
+        else this.ball.dy = -this.gameSpeed
+      }
+    }
+
+    // Calculate and save values for positions, bounds, width, height, and side
+    app.calculateBounds = function () {
+      // Save the existing width and height
+      this.previousWidth = this.width
+      this.previousHeight = this.height
+
+      // update the width and height to match the window
+      this.width = window.innerWidth
+      this.height = window.innerHeight
+
+      // Calculate the measurements of player and ball
+      this.playerWidth = this.width / 80
+      this.playerHeight = this.height / 5
+      this.ballSide = this.width / 100
+
+      // calculate the position for center and boundary
+      this.centerX = this.width / 2
+      this.centerY = this.height / 2
+      this.rightBound = this.width - this.playerWidth * 2
+      this.leftBound = this.playerWidth
+      this.topBound = 0
+      this.bottomBound = this.height - this.playerHeight
+    }
+
+    // Add the ability to draw text to the canvas
+    app.drawText = function (msg, posX, posY, alignment = "center", font = "bold 24px Arial") {
+      this.context.font = font
+      this.context.textAlign = alignment
+      this.context.fillStyle = "#900C3F"
+      this.context.fillText(msg, posX, posY)
+    }
+
+    // Handle the key press event
+    function keyDownHandler(e) {
+      if (e.keyCode == 38 || e.which == 38) {
+        app.playerRight.dy = -app.gameSpeed
+      }
+      if (e.keyCode == 40 || e.which == 40)
+        app.playerRight.dy = app.gameSpeed
+
+      if (e.keyCode == 87 || e.which == 87) {
+        app.playerLeft.dy = -app.gameSpeed
+      }
+      if (e.keyCode == 83 || e.which == 83) {
+        app.playerLeft.dy = app.gameSpeed
+      }
+
+      if (e.keyCode == 32 || e.which == 32) {
+        app.isStarted = !app.isStarted
+      }
+    }
+    app.keyDownHandler = function (e) {
+      if (e.keyCode == 38 || e.which == 38) {
+        this.playerRight.dy = -this.gameSpeed
+      }
+      if (e.keyCode == 40 || e.which == 40)
+        this.playerRight.dy = this.gameSpeed
+
+      if (e.keyCode == 87 || e.which == 87) {
+        this.playerLeft.dy = -this.gameSpeed
+      }
+      if (e.keyCode == 83 || e.which == 83) {
+        this.playerLeft.dy = this.gameSpeed
+      }
+
+      if (e.keyCode == 32 || e.which == 32) {
+        this.isStarted = !this.isStarted
+      }
+    }
```

Comment:
why do we have the same code twice?

Reply:
That's a mistake on my part. Towards the end, I refactored the key handler and resize functions and moved the logic to app.keyDownHandler and so on, that standalone function is not being used anywhere, I just forgot to delete it after refactoring. My bad!